### ISSUE TITLE
Getting different results

### ISSUE THREAD

**Issue by rank22** [rstudio/keras#301]
I am running the code from the book Deep Learning with R. The code in the second chapter gives a different values of the following quantities:
$loss
[1] 0.07519608
$acc
[1] 0.9785

My values are:
$loss
[1] 0.07382395

$acc
[1] 0.9768

Now my question is whether there could be a slight variation every time we run a code in keras?

**Comment by williamshell** [comment]
> Yes, indeed there will be a slight variation every time you run based on the random variation inherent in training neural nets. There is a way to create reproducible results, but it will be at a cost of training performance. See https://keras.rstudio.com/articles/faq.html#how-can-i-obtain-reproducible-results-using-keras-during-development

Hi, I have got the similar results #42 . No reproducible results can be obtained.  I attached the code here.  Is anyone can help me?

```
FLAGS <- flags(
  flag_numeric('dropout1', 0.4, "First Dropout Layers"),
  flag_numeric('dropout2', 0.3, "Second Dropout Layers"),
  flag_numeric('dropout3', 0.2, "Third Dropout Layers"),
  flag_integer('neurons0', 512, "first hidden layer"),
  flag_integer('neurons1', 256, "First hidden layer"),
  flag_integer('neurons2', 128, "Second hidden layer"),
  flag_integer('neurons3', 64,  "Third hidden layer"),
  flag_integer('neurons4', 32,  "Fouth hidden layer"),
  flag_integer('neurons5', 16,  "Fifth hidden layer"),
  flag_numeric('l2', 0.001,     "L2 regularization"),
  flag_numeric('lr', 0.001,     "Learning_rate")
)

tensorflow::set_random_seed(1234, disable_gpu = FALSE)
initializer_glorot_uniform(seed = 1234)
build_model <- function() {
  model <- keras_model_sequential() %>%
    layer_conv_1d(filters=32, kernel_size=2, input_shape= c(512, 1), activation="relu") %>%  
    layer_max_pooling_1d() %>%
    layer_conv_1d(filters=128, kernel_size=2, activation="relu")  %>%  
    layer_max_pooling_1d() %>%
    layer_flatten() %>%
    layer_dense  (units = FLAGS$neurons1,  kernel_initializer = "glorot_uniform", bias_initializer='zeros', activation = "relu") %>% 
    layer_dense  (units = FLAGS$neurons2,  kernel_initializer = "glorot_uniform", bias_initializer='zeros', activation = "relu") %>%
    layer_dropout(FLAGS$dropout1)%>%
    layer_dense  (units = FLAGS$neurons3,  kernel_initializer = "glorot_uniform", bias_initializer='zeros', activation = "relu") %>%
    layer_dropout(FLAGS$dropout2)%>%
    layer_dense  (units = FLAGS$neurons4,  kernel_initializer = "glorot_uniform", bias_initializer='zeros', activation = "relu") %>%
    layer_dropout(FLAGS$dropout3)%>%
    layer_dense  (units = FLAGS$neurons5,  kernel_initializer = "glorot_uniform", bias_initializer='zeros', activation = "relu") %>%
    layer_dense  (units = dim(Y_train)[2], kernel_initializer = "glorot_uniform", bias_initializer='zeros', activation = 'softmax')
  model %>% compile(
    loss       ='categorical_crossentropy',
    #    optimizer ='adam',
    optimizer = optimizer_rmsprop(learning_rate = FLAGS$lr),
    metrics   = list('accuracy')) # which metrics will be used for model valuation, and mean_absolute_error also can be used. 
  model
}
model <- build_model()
history <- model %>% fit(X_train,Y_train, validation_split = 0.10, epochs= 10, batch_size = 5, shuffle = T)
```
`sessionInfo()`

```
R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7600)

Matrix products: default

locale:
[1] LC_COLLATE=Chinese (Simplified)_China.936  LC_CTYPE=Chinese (Simplified)_China.936   
[3] LC_MONETARY=Chinese (Simplified)_China.936 LC_NUMERIC=C                              
[5] LC_TIME=Chinese (Simplified)_China.936    

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] tensorflow_2.7.0         prospectr_0.2.1          keras_2.7.0.9000         doParallel_1.0.16       
 [5] iterators_1.0.13         foreach_1.5.1            multilevelmod_0.0.0.9000 naivebayes_0.9.7        
 [9] discrim_0.1.2            stacks_0.2.0             plsmod_0.1.1             themis_0.1.4.9000       
[13] recipeselectors_0.0.1    forcats_0.5.1            stringr_1.4.0            readr_2.0.2             
[17] tidyverse_1.3.1.9000     yardstick_0.0.8.9000     workflowsets_0.1.0.9000  workflows_0.2.4         
[21] tune_0.1.6               tidyr_1.1.4              tibble_3.1.5             rsample_0.1.0           
[25] recipes_0.1.17           purrr_0.3.4              parsnip_0.1.7            modeldata_0.1.1         
[29] infer_1.0.0              ggplot2_3.3.5            dplyr_1.0.7              dials_0.0.10            
[33] scales_1.1.1.9000        broom_0.7.9              tidymodels_0.1.4.9000    pacman_0.5.1     
```
